docs/source/developer_guide/plugins/basics.rst

```diff
@@ -40,20 +40,28 @@ The plugin system was designed with the following goals in mind.
 Design guidelines
 ------------------
 
-* **Start simple.**: make use of existing classes like :py:class:`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode`, :py:class:`~aiida.orm.nodes.data.dict.Dict`, :py:class:`~aiida.orm.nodes.data.singlefile.SinglefileData`, ... Write only what is necessary to pass information from and to AiiDA.
+ * **Start simple.**
```

Comment:
Reading all of this now, it seems that this is really intended to be a guideline on how to implement a _calculation job_ but it falls under the category "plugin design guidelines". Maybe we can make this more apparent that these are guidelines when implementing a new `CalcJob` class to wrap an external code.

Reply:
Right. This falls under the general issue of https://github.com/aiidateam/aiida-core/issues/3461

I've just gone through the codebase and finally applied the naming convention we discussed there.
This has significantly increased the scope of the PR ... but it needed to be done.